src/audio/igo_nr/igo_nr.c

```diff
@@ -669,64 +563,59 @@ static void igo_nr_print_config(struct comp_dev *dev)
 		 cd->config.igo_params.nr_mode3_floor);
 	comp_dbg(dev, "  nr_mode1_pp_param53		%d",
 		 cd->config.igo_params.nr_mode1_pp_param53);
-	comp_dbg(dev, "  active_channel_idx		%d",
-		 cd->config.active_channel_idx);
+	comp_info(dev, "  active_channel_idx		%d",
+		  cd->config.active_channel_idx);
 
 }
 
-static void igo_nr_set_igo_params(struct comp_dev *dev)
+static void igo_nr_set_igo_params(struct processing_module *mod)
 {
-	struct comp_data *cd = comp_get_drvdata(dev);
+	struct comp_data *cd = module_get_private_data(mod);
 	struct sof_igo_nr_config *p_config = comp_get_data_blob(cd->model_handler, NULL, NULL);
+	struct comp_dev *dev = mod->dev;
 
 	comp_info(dev, "igo_nr_set_igo_params()");
 	igo_nr_check_config_validity(dev, cd);
 
 	if (p_config) {
 		comp_info(dev, "New config detected.");
 		cd->config = *p_config;
-		igo_nr_print_config(dev);
+		igo_nr_print_config(mod);
 	}
 }
 
 /* copy and process stream data from source to sink buffers */
-static int32_t igo_nr_copy(struct comp_dev *dev)
+static int igo_nr_process(struct processing_module *mod,
+			  struct input_stream_buffer *input_buffers, int num_input_buffers,
+			  struct output_stream_buffer *output_buffers, int num_output_buffers)
 {
-	struct comp_copy_limits cl;
-	struct comp_buffer *sourceb, *sinkb;
-	struct comp_data *cd = comp_get_drvdata(dev);
-	int32_t src_frames;
-	int32_t sink_frames;
+	struct comp_dev *dev = mod->dev;
+	struct comp_data *cd = module_get_private_data(mod);
+	struct audio_stream *source = input_buffers[0].data;
+	struct audio_stream *sink = output_buffers[0].data;
+	int frames = input_buffers[0].size;
 
 	comp_dbg(dev, "igo_nr_copy()");
 
-	sourceb = list_first_item(&dev->bsource_list, struct comp_buffer,
-				  sink_list);
-	sinkb = list_first_item(&dev->bsink_list, struct comp_buffer,
-				source_list);
-
 	/* Check for changed configuration */
 	if (comp_is_new_data_blob_available(cd->model_handler))
-		igo_nr_set_igo_params(dev);
-
-	/* Get source, sink, number of frames etc. to process. */
-	comp_get_copy_limits(sourceb, sinkb, &cl);
+		igo_nr_set_igo_params(mod);
```

Comment:
possible put set params to set_config? this is time critical place

Reply:
No, let's not remove and not change the feature. The IPC4 handler should handle the binary blob as ipc3 does.